### ISSUE TITLE
WP_CLI\Core\CoreUpgrader::download_package needs to add $hook_extra parameter

### ISSUE THREAD

**Issue by afragen** [wp-cli/core-command#167]
## Bug Report

--- ✅ If you are in the correct location now... --->

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).
- [x] Yes, more specifically, I reviewed the guidelines on [how to write clear bug reports](https://make.wordpress.org/cli/handbook/bug-reports/).

**Describe the current, buggy behavior**

In WP 5.5 an additional parameter was added to WP_Upgrader::download_package( $package, $check_signatures, $hook_extra ). When trying to force an update to a previous verison eg `wp core update --force --version=beta1` a PHP warning about the difference between WP_CLI\Core\CoreUpgrader::download_package and WP_Upgrader::download_package.

This is using WP-CLI 2.5.0-alpha

> PHP Warning:  Declaration of WP_CLI\Core\CoreUpgrader::download_package($package, $check_signatures = true) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///Applications/Local Beta.app/Contents/Resources/extraResources/bin/wp-cli/wp-cli.phar/vendor/wp-cli/core-command/src/WP_CLI/Core/CoreUpgrader.php on line 30

> Downloading update from https://wordpress.org/wordpress-beta1.zip...
> Error: Download failed.

**Describe how other contributors can replicate this bug**

`wp core update --force --version=beta1`

**Describe what you would expect as the correct outcome**

Update occurs.

**Let us know what environment you are running this on**

```
OS:	Darwin 19.6.0 Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64
Shell:	/bin/zsh
PHP binary:	/Users/afragen/Library/Application Support/Local Beta/lightning-services/php-7.4.1+12/bin/darwin/bin/php
PHP version:	7.4.1
php.ini used:	/Users/afragen/Library/Application Support/Local Beta/run/HJwKJqoPU/conf/php/php.ini
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/Users/afragen/Local_Beta_Sites/lightning/app/public
WP-CLI packages dir:	
WP-CLI global config:	/Applications/Local Beta.app/Contents/Resources/extraResources/bin/wp-cli/config.yaml
WP-CLI project config:	
WP-CLI version:	2.5.0-alpha
```

**Comment by kkmuffme** [comment]
Issue is reproducable for us too

**Comment by strarsis** [comment]
Issue is fixed in WP CLI nightly release (`2.5.x`).
`wp cli update --nightly` (not just on your workstation - but also on the actual server!)
